src/macaron/slsa_analyzer/git_url.py

```diff
@@ -25,22 +25,126 @@
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "", offline_mode: bool = False) -> bool:
+def parse_git_branch_output(content: str) -> list[str]:
+    """Return a list of branch name from the output of ``git branch``.
+
+    This function only work for the output format of the first form of ``git branch`` where the branches are listed.
+    See https://git-scm.com/docs/git-branch for definition of the first form of ``git branch``.
+
+    Parameters
+    ----------
+    content : str
+        The raw output as string from the ``git branch`` command.
+
+    Returns
+    -------
+    list[str]
+        The list of strings where each string is a branch element from the raw output.
+
+    Examples
+    --------
+    >>> print(content)
+    * (HEAD detached at 7fc81f8)
+      master
+      remotes/origin/HEAD -> origin/master
+      remotes/origin/master
+      remotes/origin/v2.dev
+      remotes/origin/v3.dev
+    >>> _parse_branch_output(content)
+    [
+        '(HEAD detached at 7fc81f8)',
+        'master',
+        'remotes/origin/HEAD -> origin/master',
+        'remotes/origin/master',
+        'remotes/origin/v2.dev',
+        'remotes/origin/v3.dev',
+    ]
+    """
+    raw_branch_names = content.split("\n")
+    result = []
+    for name in raw_branch_names:
+        # Ignore elements that doesn't contain only whitespaces. This is because the raw content of git branch
+        # can have extra new line at the end, which can be picked up as an empty element in `raw_branch_names`.
+        if len(name.strip()) == 0:
+            continue
+
+        # The ``*`` symbol will appear next to the branch name where HEAD is currently on.
+        # Branches in git cannot have ``*`` in its name so we can safely replace without tampering with its actual name.
+        # https://git-scm.com/docs/git-check-ref-format
+        cleaned_up_branch_name = name.replace("*", "").strip()
+
+        result.append(cleaned_up_branch_name)
```

Comment:
Nit: Just some naming suggestions:
`raw_branch_names -> git_branch_output_lines`
`result -> branches`
`name -> line`
`cleaned_up_branch_name` -> `branch`

Reply:
Fixed in https://github.com/oracle/macaron/pull/548/commits/30665ed12f2842bb4f1e4c6c48ccd119f1858f60